### DISCUSSION TITLE
KDE file picker and other inegration

### DISCUSSION THREAD

**Discussion by ghost** [Floorp-Projects/Floorp#679] [Feature Request]
Not sure if in scope but there are patches to make firefox integrate better with KDE, which out of the box it basically doesn't do at all.

https://gitlab.com/Thaodan/firefox-kde-opensuse

Also a breeze style exists:

https://github.com/mirkobrombin/firefox-breeze-theme

**Comment by tannenfreund87** [comment]
You can set up xdg-desktop-portal file picker and mime-types in about:config. That will use the kde file picker in plasma, and the gnome file picker on gnome (yay, finally thumbnails in a grid, after 3 decades!). Firefox will use that eventually as default. Maybe floorp can switch to using portals in the linux version by default already?

**Comment by tomdavidson** [comment]
Lots can be done through config and extensions but functions like global menu support needs the "opensuse patch". At that point the rest of the integration can be included behind the "is KDE" conditional.  

Im for porting the open source kde patches to Floorp and would be happy to do the work as a mentee.

**Reply by CutterKnife** [reply]
If you have a feature that you think you could implement yourself, please send a pull request to Floorp-core or Floorp :)

**Reply by tomdavidson** [reply]
I should have placed emphasis on 'as a mentee' :D  but yes, I would like to port the OpenSUSE KDE integration patches to Floorp.  Is there an existing contributor that can provide a tour or help make a plan of how to solve?  For example, I had expected several *.patch files in the repro and not sure how to test the new functionality.

Id be happy to open a draft PR to ask for help on outlining what the PR should do.
